### PR TITLE
Move govTokenID to global state

### PR DESCRIPTION
**Pull request by sczembor**
Closes #x

## Proposed Changes

- Updated dao smart contract template by moving the `gov_token_id` from template params to global state
- Updated the tests

### CODE REVIEW

File: examples/dao/scripts/deploy.js [hunk 1]
```diff
@@ -41,14 +43,13 @@ async function run(runtimeEnv, deployer) {
...
+			globalInts: 5,
 			globalBytes: 2,
 			appArgs: appArgs,
 		},
-		{},
-		templateParam
+		{}
 	);
-	console.log(daoAppInfo);
+	console.log("after deployment", daoAppInfo);
```
Comment (sczembor) [inline examples/dao/scripts/deploy.js:52]:
```suggestion
	console.log(daoAppInfo);
```

File: examples/dao/scripts/deploy.js [hunk 2]
```diff
@@ -29,8 +30,9 @@ async function run(runtimeEnv, deployer) {
 		`int:${maxDuration}`,
 		`str:${url}`,
 		`str:${daoName}`,
+		`int:${govTokenId}`,
 	];
-	const templateParam = { ARG_GOV_TOKEN: govToken.assetIndex };
+	// const templateParam = { ARG_GOV_TOKEN: govToken.assetIndex };
```
Comment (sczembor) [inline examples/dao/scripts/deploy.js:35]:
```suggestion
```

File: examples/dao/assets/dao-app-approval.py [hunk 3]
```diff
@@ -543,13 +545,13 @@ def scratch_proposal_active(idx: Int):
...
+    # params = {
+    #     "ARG_GOV_TOKEN": 99
+    # }
 
-    # Overwrite params if sys.argv[1] is passed
-    if(len(sys.argv) > 1):
-        params = parse_params(sys.argv[1], params)
+    # # Overwrite params if sys.argv[1] is passed
+    # if(len(sys.argv) > 1):
+    #     params = parse_params(sys.argv[1], params)
```
Comment (sczembor) [inline examples/dao/assets/dao-app-approval.py:554]:
```suggestion
```

File: examples/dao/assets/dao-app-approval.py [hunk 4]
```diff
@@ -543,13 +545,13 @@ def scratch_proposal_active(idx: Int):
     return program
 
 if __name__ == "__main__":
-    params = {
-        "ARG_GOV_TOKEN": 99
-    }
+    # params = {
+    #     "ARG_GOV_TOKEN": 99
+    # }
```
Comment (sczembor) [inline examples/dao/assets/dao-app-approval.py:550]:
```suggestion
```

File: examples/dao/scripts/deploy.js [hunk 5]
```diff
@@ -29,8 +30,8 @@ async function run(runtimeEnv, deployer) {
 		`int:${maxDuration}`,
 		`str:${url}`,
 		`str:${daoName}`,
+		`int:${govTokenId}`,
```
Comment (vuvoth) [inline examples/dao/scripts/deploy.js:33]: I think at here the correct value is `govToken.assetIndex`

File: examples/dao/assets/dao-app-approval.py [hunk 6]
```diff
@@ -1,10 +1,9 @@
...
 sys.path.insert(0,'..')
 
-from algobpy.parse import parse_params
 from pyteal import *
 
-def approval_program(ARG_GOV_TOKEN):
+def approval_program():
     """
     A stateful app with governance rules. Stores
     deposit, min_support, min_duration, max_duration, url, dao_name.
```
Comment (vuvoth) [inline examples/dao/assets/dao-app-approval.py:9]:
```suggestion
    deposit, min_support, min_duration, max_duration, url, dao_name, gov_token_id.
```

**Review by vuvoth — COMMENTED** [review]
Nice work! I left few comments. Thanks!

File: examples/dao/scripts/deploy.js (hunk 5 above)
Comment (sczembor) [inline examples/dao/scripts/deploy.js:33]:
```suggestion
		`int:${govToken.assetID}`,
```

File: examples/dao/scripts/deploy.js [hunk 7]
```diff
@@ -21,6 +21,7 @@ async function run(runtimeEnv, deployer) {
 	const maxDuration = 5 * 60; // 5min (maximum voting time in number of seconds)
 	const url = "www.my-url.com";
 	const daoName = "DAO";
+	const govTokenId = 99;
```
Comment (sczembor) [inline examples/dao/scripts/deploy.js:24]:
```suggestion
```

File: examples/dao/scripts/deploy.js (hunk 5 above)
Comment (sczembor) [reply examples/dao/scripts/deploy.js]: you are right

File: CHANGELOG.md [hunk 8]
```diff
@@ -75,6 +75,7 @@ Dependencies:
 
 - We updated the examples/DAO design. We removed treasury Smart Signature to simplify deposit management. Now a DAO app is managing voting, deposits and treasury.
 - Enabled PyTEAL Optimizer option in all our examples.
+- DAO template has been improved by moving the `gov_token_id` from template parameter to the global state.
```
Comment (ratik21) [inline CHANGELOG.md:78]: We can also specify the reason. i.e the bytecode remains the same after each deploy

File: examples/dao/assets/dao-app-approval.py [hunk 9]
```diff
@@ -543,13 +545,7 @@ def scratch_proposal_active(idx: Int):
...
-        "ARG_GOV_TOKEN": 99
-    }
 
-    # Overwrite params if sys.argv[1] is passed
-    if(len(sys.argv) > 1):
-        params = parse_params(sys.argv[1], params)
 
     optimize_options = OptimizeOptions(scratch_slots=True)
-    print(compileTeal(approval_program(params["ARG_GOV_TOKEN"]), Mode.Application, version = 5, optimize=optimize_options))
+    print(compileTeal(approval_program(), Mode.Application, version = 5, optimize=optimize_options))
```
Comment (ratik21) [inline examples/dao/assets/dao-app-approval.py:551]: Can we upgrade the version as well? (to `6`)?

**Review by ratik21 — APPROVED** [review]

**Review by vuvoth — APPROVED** [review]

**Pull request merged**